### PR TITLE
Adds :z option to support SeLinux

### LINKED ISSUE #39: drop selinux code in favour of `:z`?
Currently, before starting the container we look at the selinux status with https://github.com/ceph/cn/blob/249e38107c8ae0b241b2f80311545258aa3935b0/cmd/utils.go#L56-L96. This is used when using a dedicated directory for OSD data store: https://github.com/ceph/cn/blob/249e38107c8ae0b241b2f80311545258aa3935b0/cmd/start.go#L185

The docker CLI has a `:z` arg when bindmounting directory in a containers (see https://www.projectatomic.io/blog/2015/06/using-volumes-with-docker-can-cause-problems-with-selinux/). We need to find the `:z` equivalent from the Docker API and use it instead calling raw commands.

### PR DESCRIPTION
**Pull request by deep-j**
This commit makes use of  ':z' option on volume mounts when SeLinux is Enforced to tell Docker to relabel file object on the shared volumes.
Resolves #39

### CODE REVIEW

**Comment by leseb** [comment]
@deep-j why did you close your PR?

**Comment by deep-j** [comment]
I was checking if Selinux is set to enforcing and then applied the z flag.
But the z flag can be added otherwise too.
So I will redo the commit and send a fresh PR.

**Comment by leseb** [comment]
You don't need to close the PR, you can always change your commit history and push force your branch.

**Comment by deep-j** [comment]
Removed check for SeLinux while adding z option on volume mounts.

File: cmd/start.go [hunk 1]
```diff
@@ -181,11 +181,10 @@ func runContainer(cmd *cobra.Command, args []string) {
 			if !testEmptyDir {
 				log.Fatal(getUnderlyingStorage(flavor) + " is not empty, doing nothing.")
 			}
-			if runtime.GOOS == "linux" {
```
Comment (leseb) [inline cmd/start.go:184]: Why did you remove this check? This Selinux thing is onlu valid on Linux.

**Review by leseb — CHANGES_REQUESTED** [review]

File: cmd/start.go (hunk 1 above)
Comment (deep-j) [reply cmd/start.go:184]: Added the check for linux environment .

File: cmd/start.go [hunk 2]
```diff
@@ -181,11 +181,13 @@ func runContainer(cmd *cobra.Command, args []string) {
...
-				applySeLinuxLabel(getUnderlyingStorage(flavor))
-			}
 			envs = append(envs, "OSD_PATH="+getUnderlyingStorage(flavor))
-			volumeBindings = append(volumeBindings, getUnderlyingStorage(flavor)+":"+getUnderlyingStorage(flavor))
+			if runtime.GOOS == "linux"  {
```
Comment (leseb) [inline cmd/start.go:186]: You need to keep `volumeBindings` outside of the Linux condition and only add `:z` if Linux. Btw did you actually validated that adding `:z` works?

**Review by leseb — CHANGES_REQUESTED** [review]

File: cmd/start.go (hunk 2 above)
Comment (deep-j) [reply cmd/start.go:186]: Yes I have validated that adding `:z` works in selinux enforced environment.
The functional_tests of the build is failing due to [https://github.com/ceph/ceph-container/issues/1293]( https://github.com/ceph/ceph-container/issues/1293) for the ceph/daemon:lastest image which is used as default image in ceph-nano.

**Review by leseb — APPROVED** [review]
